### ISSUE TITLE
How to Paraphrase with GPT2?

### ISSUE THREAD

**Issue by zanderbush** [huggingface/transformers#4965]
# ❓ Questions & Help

**Comment by patil-suraj** [comment]
You should rather use a seq2seq model for paraphrasing like T5 or BART. But if you want to do it using GPT-2 then maybe you can use this format

`input: input_text paraphrase: parahrase_text`

 while training,  set attention mask to 0 on the paraphrased text

and when generating just pass `input: input_text paraphrase: `  and sample till the `eos` token

**Comment by zanderbush** [comment]
Thank you. I'll be sure to try that! I know there are datasets online that I can use, but are there edits I should make to ensure that I get a suitable output? I've read something about cheating, and I want to avoid that. (I'm sorry if I am using the wrong terminology)

**Comment by shamoons** [comment]
Can we use `T5` or `BART` like we would `GPT-2`?

**Comment by patil-suraj** [comment]
hi @shamoons , yes you can, have a look at this https://madewithml.com/projects/1094/paraphrase-any-question-with-t5-text-to-text-transformer/

**Comment by shamoons** [comment]
This looks like it’s only for questions. What about arbitrary sentences?

**Comment by zanderbush** [comment]
> This looks like it’s only for questions. What about arbitrary sentences?

You can do it with arbitrary sentences as well, but you'll need to fine-tune it yourself.

**Comment by kingglory** [comment]
> hi @shamoons , yes you can, have a look at this https://madewithml.com/projects/1094/paraphrase-any-question-with-t5-text-to-text-transformer/
This url is gone.

**Comment by patil-suraj** [comment]
@kingglory 

This is the [url](https://github.com/ramsrigouthamg/Paraphrase-any-question-with-T5-Text-To-Text-Transfer-Transformer-) for the repo of that project.

Also there are quite a few paraphrase models on the [hub](https://huggingface.co/models?search=paraphrase) that you can try
